db/file-queries.js

```diff
@@ -5,7 +5,13 @@ var db = require('./db-connect')
  * All DB actions for LinksUser.
  */
 function getFileForUser (username, tutorialId) {
-  return db.one('select "data" from "LinksFile" where "username" = $1 and "tutorial_id"=$2', [username, tutorialId])
+  return db.task('getFileForUSer', t => {
+    return db.oneOrNone('SELECT "data" FROM "LinksFile" WHERE "username" = $1 AND "tutorial_id"=$2', [username, tutorialId])
+      .then(file => {
+        return file || t.one('INSERT INTO "LinksFile" ("data", "tutorial_id", "username") VALUES' +
+          '((SELECT "source" FROM "LinksTutorial" WHERE "tutorial_id" = $1), $1, $2) RETURNING "data"', [tutorialId, username])
```

Comment:
Since you are only returning the `data` field from DB, instead of the whole row, this function should be renamed into something like `getFileSourceForUser` or `getTutorialSourceForUser`.